pylint/checkers/variables.py

```diff
@@ -1641,6 +1642,10 @@ def _check_consumer(
                 if not (
                     self._postponed_evaluation_enabled
                     and isinstance(stmt, (nodes.AnnAssign, nodes.FunctionDef))
+                ) and not (
+                    PY310_PLUS
+                    and isinstance(stmt, nodes.Match)
+                    and isinstance(node.parent, nodes.Compare)
```

Comment:
This doesn't solve this false positive, because the `.parent` is `nodes.Name`:
```python
match ("example", "one"):
    case (x, y) if x:  # <-- pylint complains about 'x' here
        print("woo!")
    case _:
        print("boo...")
```